tests/middleware_tests/test_names.py

```diff
@@ -860,6 +860,27 @@ def test_split_name_into_parts(name, expected_as_dict, strict):
     assert result == expected
 
 
+@pytest.mark.parametrize(
+    "name, expected_as_dict", REGULAR_NAME_PARTS_PARSING_TEST_CASES
+)
+@pytest.mark.parametrize("strict", [True, False], ids=["strict", "non-strict"])
+def test_merge_last_name_first_inverse(name, expected_as_dict, strict):
+    """Tests that merging name parts using the last-name-first method
+    maintains the "semantics" of the name.
+
+    This property does not hold for certain values that contain '\\'.
+    """
+
+    # skip cases with \
+    if any("\\" in "".join(name) for name in expected_as_dict.values()):
```

Comment:
Can we somehow document in which cases we cannot handle this? I have struggle understanding the severity of this case.

Reply:
As far as I can tell, the cases that cannot be handled are those where either the last name or "von" part ends with an odd number of `\`, since in those cases the `,` is escaped when the name parts are put back together. I add an extra `\` in those cases, which I _think_ preserves the semantics of the name, though I'm not 100% sure.

The filter here is more aggressive than what I put above though, to keep it simple. But I guess it's better if the filter captures exactly the cases for which the "inverse" property holds?

Where would you like the documentation for the edge cases?

Reply:
Ok, I guess that's enough of an edge case and ok to have it in the code, since I really do not think that single trailing `\`  are common enough to block this PR :-)

Reply:
```suggestion
    # cases where either the last name or "von" part ends with an odd number of `\` cannot be handled, 
    # since in those cases the `,` is escaped when the name parts are put back together
    if any("\\" in "".join(name) for name in expected_as_dict.values()):
```

Reply:
Thanks. I've added the comment and made the filter a bit more lenient so only cases with an odd number of trailing `\` are discarded.